common/src/project.rs

```diff
@@ -41,14 +47,26 @@ impl std::fmt::Display for ProjectName {
 
 impl ProjectName {
     pub fn is_valid(hostname: &str) -> bool {
+
         fn is_valid_char(byte: u8) -> bool {
             (b'a'..=b'z').contains(&byte)
                 || (b'A'..=b'Z').contains(&byte)
                 || (b'0'..=b'9').contains(&byte)
                 || byte == b'-'
         }
 
-        !(hostname.bytes().any(|byte| !is_valid_char(byte))
+        fn is_profanity_free_and_not_reserved(hostname: &str) -> bool {
+
+            let mut reserved_words = HashSet::new();
+            // Reserved words can be easily added here. 
+            reserved_words.insert("shuttle.rs".to_string());
```

Comment:
Creating the `HashSet` from an array ([the last doc example](https://doc.rust-lang.org/std/collections/struct.HashSet.html)) may make it even easier to add reserve words in the future :man_shrugging: